### ISSUE TITLE
stopped working today

### ISSUE THREAD

**Issue by Rowsol** [pietervanheijningen/clickbait-remover-for-youtube#56]
it still technically works, but it requires that you toggle it off and on each time you visit youtube.

**Comment by pietervanheijningen** [comment]
Hey! For me it still works. So it's probably an A/B test from youtube, which makes this a bit hard to test.

Can you make a screenshot of the HTML of the thumbnail that has not changed? Like so: with inspect element

![image](https://github.com/pietervanheijningen/clickbait-remover-for-youtube/assets/28759591/9ccf8cba-632a-407e-bc6c-25e3aa338315)

Then i should be able to figure out why it is not changing for you. Probably the src url in the img tag is different on your youtube version.

Thanks!

**Comment by Rowsol** [comment]
I hope this helps
![Screenshot (348)](https://github.com/pietervanheijningen/clickbait-remover-for-youtube/assets/35880360/fba229ac-e0f0-411c-b7c9-f57499714667)

**Comment by pietervanheijningen** [comment]
Sorry did not explain it clearly, can you also expand the yt-image HTML tag? So I can see the actual image URL? like on my screenshot

**Comment by Rowsol** [comment]
I hope this is right, this stuff is a bit over my head.

![Screenshot (349)](https://github.com/pietervanheijningen/clickbait-remover-for-youtube/assets/35880360/2215e16e-455f-4194-89c1-d5a5101b7569)

**Comment by theshatterstone** [comment]
Same here. Titles still work, but the thumbnails don't until I set the toggles again. Here's the picture
![Screenshot_2023-06-04_20-50-59](https://github.com/pietervanheijningen/clickbait-remover-for-youtube/assets/90517516/6bc3d8a0-5d5d-4123-9fea-f8cc139ea472)

Also, I just noticed something. I use Firefox as well and it works perfectly on Firefox. All the screenshots posted so far are from Brave browser. It seems it might be an issue stemming from either Chromium-based browser, or from Brave itself.

**Comment by Rowsol** [comment]
Yeah, it appears to be an issue with Brave. Thanks for the heads up.

**Comment by Izofeu** [comment]
I found a dirty fix for impatient people like me:
Edit the extension - file youtube.js:
Add
`function fixfix()
{
	updateThumbnails("hq2");
}

setInterval(fixfix, 1000);`
somewhere in the code and run. Dirty and bad fix, but at least makes Youtube usable.

**Comment by Rowsol** [comment]
> I found a dirty fix for impatient people like me: Edit the extension - file youtube.js: Add
> 

Can you elaborate? Where do you get this extension?

**Comment by Izofeu** [comment]
1. Go to %localappdata%\BraveSoftware\Brave-Browser\User Data
2. Look for "omoinegiohhgbikclijaniebjpkeopip"
3. Open the folder, copy 0.8_0 folder onto somewhere else
4. Remove the extension from Brave
5. Delete _metadata folder from the copied folder
6. Open 0.8_0\js\youtube.js
7. Paste the code from above somewhere into the file, I pasted it here (a fragment of code was provided):
```
chrome.runtime.onMessage.addListener(function (message) {
        Object.keys(message).forEach(function (change) {
            switch (change) {
                case 'preferred_thumbnail_file':
                    updateThumbnails(
                        message[change].newValue === undefined ? 'hq1' : message[change].newValue
                    );
                    break;
                case 'video_title_format':
                    updateCSS(message[change].newValue);
                    break;
            }
        })
    });
	//
        // I PASTED THE CODE HERE
        //
	function fixfix()
	{
                // REPLACE HQ2 WITH TYPE OF THUMBNAIL YOU WANT - hq1 START, hq2 MIDDLE, hq3 END
		updateThumbnails("hq2");
	}
	
	setInterval(fixfix, 1000);

        // 
        // END OF PASTED CODE
        // 

    function updateCSS(option) {

        let appendingElement = false;

        if (window.styleElement === null) {
            appendingElement = true;
            window.styleElement = document.createElement('style');
        }
```

8. Open Extensions in Brave
9. Enable Developer mode in top right
10. Click load unpacked
11. Select the 0.8_0 folder (that has manifest.json in it)
12. Click load
We need to do all these steps as extensions are signed and altering the code of signed extensions will prevent them from running.

**Comment by Rowsol** [comment]
Very much appreciated!

**Comment by pietervanheijningen** [comment]
Hi everyone, thanks for the temporary fix @Izofeu. But a proper fix should be possible, likely just something has to be changed in how the thumbnail url listeners are setup. Specifically here: https://github.com/pietervanheijningen/clickbait-remover-for-youtube/blob/72a37edf345999d73f67486e3a1431058de18638/js/background.js#L51

Unfortunately the extension still works fine for me, this is probably due to an A/B test from youtube or chrome, which is why it is broken for some people. Strangely enough the URL to the thumbnail still seems to be the same, which is why I asked for the screenshots. So i'm a bit stumped on what it could possibly be, for me its just a guessing game.

Hoping someone who has this A/B test can figure it out and submit a pull request.

**Comment by I-am-the-Users** [comment]
The problem is fixed after deleting these two folders.
`%localappdata%\BraveSoftware\Brave-Browser\User Data\Default\DNR Extension Rules\omoinegiohhgbikclijaniebjpkeopip`
`%localappdata%\BraveSoftware\Brave-Browser\User Data\Default\Sync Extension Settings\omoinegiohhgbikclijaniebjpkeopip`

**Comment by andrzejgrzelak** [comment]
I looked into this and I'm pretty sure it's connected to this issue:
https://github.com/brave/brave-browser/issues/30854

I can confirm that if you disable&enable the extension it works until the browser restart

**Comment by pietervanheijningen** [comment]
> I looked into this and I'm pretty sure it's connected to this issue: [brave/brave-browser#30854](https://github.com/brave/brave-browser/issues/30854)
> 
> I can confirm that if you disable&enable the extension it works until the browser restart

Thanks for this finding! Almost certain that that is the issue then. Hope brave fixes it soon, Kind of odd that brave deviates from standard chromium behaviour. Don't really feel like I am doing very advanced/weird things with my extension.

**Comment by Mhowser** [comment]
FYI [DeArrow](https://dearrow.ajay.app) works on brave, I've been testing it out and I think I'm gonna just replace CBR with DeArrow.

**Comment by jrejaud** [comment]
> FYI [DeArrow](https://dearrow.ajay.app) works on brave, I've been testing it out and I think I'm gonna just replace CBR with DeArrow.

Second DeArrow. I am a big fan of CBR but I think DeArrow is a bit better, total clickbait death is non-negotiable for me.

**Comment by pietervanheijningen** [comment]
appears to be fixed in the latest brave update according to https://github.com/pietervanheijningen/clickbait-remover-for-youtube/issues/66?
